src/ConfigDefinition.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Processor\FlattenFolders;
+
+use Keboola\Component\Config\BaseConfigDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+/**
+ * Class ConfigDefinition
+ *
+ * ConfigDefinition for the processor
+ *
+ * @package Keboola\Processor\FlattenFolders
+ */
+class ConfigDefinition extends BaseConfigDefinition
+{
+    /**
+     * @return \Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition|\Symfony\Component\Config\Definition\Builder\NodeDefinition
+     */
+    public function getParametersDefinition()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root("parameters");
+        // @formatter:off
+        $rootNode
+            ->children()
+                ->integerNode("depth")
+                    ->min(0)
+                    ->max(1)
```

Comment:
ha, jo vidim, max depth 1

Reply:
jj, přesně, v `/data/out/tables` můžou bejt složky a všechny soubory v těch složkách se pak importují do stejný tabulky.

Reply:
Viděl jsem použití třeba tak, že stahnu nějakou strukturu z FTP (/report/2017/01/12/report.csv) a pak to chci nahrát do kebooly, ale ta neumí složky, takže je potřeba to flattnout

Reply:
Jo, to je dobrej usecase, na ten jsem taky myslel, depth 0/1 slouží jen k tomu, jestli to má zachovat první úroveň adresářů v `data/out/files|tables` (ta slouží pro slicovaný soubory/tabulky).

Reply:
Současnej S3 extraktor už tohl, co popisuješ, dělá sám, tak tam defaultně není potřeba (a měnit ho by byl BC break, tak tam už to zůstane). Nový komponenty už by nic takovýho řešit neměly.